imagine/simulators/simulator.py

```diff
@@ -251,8 +251,8 @@ def _find_dependencies(self, fields):
                     dependencies[field].update(fdep)
 
         # Subsititutes any field type string by field classes
-        for k, deps in dependencies.items():
-            for dep in tuple(deps):
+        for deps in dependencies.values():
+            for dep in deps:
```

Comment:
This change leads to an error in the tests:
```
>           for dep in deps:
E           RuntimeError: Set changed size during iteration
```
The previous implementation created a tuple with `tuple(deps)` to avoid this kind of error.

Reply:
Uhm, yeah, you are totally right.
I missed that one.